src/client/bb-client.js

```diff
@@ -275,13 +275,16 @@ BBClient.ready = function(input, callback, errback){
     // Check if 2 minutes from access token expiration timestamp
     var tokenResponse = getPreviousToken();
     var payloadCheck = jwt.decode(tokenResponse.access_token);
-    var nearExpTime = Math.floor(Date.now() / 1000) >= (payloadCheck['exp'] - 120);
-
-    if (tokenResponse.refresh_token
-      && tokenResponse.scope.indexOf('online_access') > -1
-      && nearExpTime) { // refresh token flow
-      accessTokenResolver = completeTokenRefreshFlow();
-    } else { // existing access token flow
+    if (payloadCheck && payloadCheck['exp']) {
```

Comment:
I'm wondering why the non-oauth flow is coming in the validTokenResponse() case in the first place.  Can we somehow bypass this, if it's not needed?

Reply:
Added a check for a fake access token (non-oauth) before handling other valid token logic.
https://github.com/cerner/client-js/pull/4/commits/fe9dee21d1d7bdac4a092179a486dba5b5627cb8